gradle/dependencies.gradle

```diff
@@ -15,367 +15,386 @@
-                            changeVersionInMetadata it, '2.11.0'
-                        }
-                    }
-                }
-            }
-            withModule('org.eclipse.jgit:org.eclipse.jgit') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group == 'com.googlecode.javaewah' }}
-                }
-            }
-            withModule('com.amazonaws:aws-java-sdk-core') {
-                withVariant('default') {
-                    withDependencies {
-                        it.removeAll { it.name == 'jackson-dataformat-cbor' || it.group == 'commons-logging' }
-                        it.findAll { it.name == 'httpclient' }.each { changeVersionInMetadata it, versions.httpclient }
-                    }
-                }
-                withVariant('runtime') {
-                    withDependencies {
-                        it.removeAll { it.name == 'jackson-dataformat-cbor' || it.group == 'commons-logging' }
-                        it.findAll { it.name == 'httpclient' }.each { changeVersionInMetadata it, versions.httpclient }
-                    }
+                allVariants {
+                    withDependencies { it.removeAll { it.group == 'org.sonatype.sisu' } }
                 }
             }
-            withModule('dom4j:dom4j') {
-                withVariant('runtime') {
-                    withDependencies { it.removeAll { it.group == 'xml-apis' }}
-                }
-            }
-            withModule('com.google.code.findbugs:findbugs') {
-                withVariant('default') {
-                    withDependencies { it.removeAll { it.group != 'dom4j' }}
+
+            // Gradle distribution - replace similar library with different coordinates
+            replaceGoogleCollectionsWithGuava(handler, 'org.codehaus.plexus:plexus-container-default')
+
+            replaceJunitDepWithJunit(handler, 'org.jmock:jmock-junit4')
+
+            replaceBeanshellWithApacheBeanshell(handler, 'org.testng:testng')
+
+            replaceLog4JWithAdapter(handler, 'org.apache.xbean:xbean-reflect')
+
+            replaceJCLWithAdapter(handler, 'org.apache.xbean:xbean-reflect')
+            replaceJCLWithAdapter(handler, 'org.apache.httpcomponents:httpclient')
+            replaceJCLWithAdapter(handler, 'com.amazonaws:aws-java-sdk-core')
+            replaceJCLWithAdapter(handler, 'org.apache.httpcomponents:httpmime')
+            replaceJCLWithAdapter(handler, 'net.sourceforge.htmlunit:htmlunit')
+            replaceJCLWithAdapter(handler, 'org.apache.maven.wagon:wagon-http')
+            replaceJCLWithAdapter(handler, 'org.apache.maven.wagon:wagon-http-shared4')
+
+            replaceJCLConstraintWithAdapter(handler, 'org.codehaus.groovy:groovy-all')
+
+            //TODO check if we can upgrade the following dependencies and remove the rules
+            downgradeIvy(handler, 'org.codehaus.groovy:groovy-all')
```

Comment:
I think Nebula wanted the ability to add a constraint that downgraded a transitive version. Let's consider how that could be possible, to avoid requiring rules like this.